.github/workflows/ci.yml

```diff
@@ -0,0 +1,71 @@
+name: CI
+
+on:
+  # Run when main gets updated, i.e. PR merge
+  push:
+    branches: ["main"]
+
+  # Run when a PR is opened, and whenever new commits are added to the PR
+  pull_request:
+
+  # Run when manually triggered
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    permissions:
+      # Allow GitHub to request an OIDC JWT ID token, to use with aws-actions/configure-aws-credentials
+      id-token: write
+      contents: read
+
+    steps:
+      # Checkout the repository
+      - uses: actions/checkout@v3
```

Comment:
There's a [v4](https://github.com/actions/checkout/releases) of this action.

Reply:
thanks - updated to v4